stl/inc/xstddef

```diff
@@ -41,6 +41,30 @@ struct binary_function { // base class for binary functions
 };
 #endif // _HAS_AUTO_PTR_ETC
 
+#pragma warning(push)
+#pragma warning(disable : 4242) // '%s': conversion from '%s' to '%s', possible loss of data (/Wall)
+#pragma warning(disable : 4244) // '%s': conversion from '%s' to '%s', possible loss of data (Yes, duplicated message.)
+#pragma warning(disable : 4365) // '%s': conversion from '%s' to '%s', signed/unsigned mismatch (/Wall)
+#pragma warning(disable : 5215) // '%s' a function parameter with a volatile qualified type is deprecated in C++20
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-volatile"
+#endif // __clang__
+
+template <class _Ty>
+_NODISCARD _Ty _Fake_copy_init(_Ty) noexcept;
+// _Fake_copy_init<T>(E):
+// (1) has type T [decay_t<decltype((E))> if T is deduced],
+// (2) is well-formed if and only if E is implicitly convertible to T and T is destructible, and
+// (3) is non-throwing if and only if both conversion from decltype((E)) to T and destruction of T are non-throwing.

```

Comment:
Because `_Fake_decay_copy` and `_Implicitly_convert_to` both imply what they're used for - `_Fake_decay_copy` is a "fake" version of the standard's `decay_copy`:

```cxx
template <class T>
typename decay<T>::type decay_copy(T&& v);
```

while `_Implicitly_convert_to` implies that a template type parameter is required; `_Implicitly_convert_to(blah)` just looks incorrect.

`_Fake_copy_init`, on the other hand, does not have either implication, and it tells you exactly what it does - copy initialization.